tests/unit/test_messages.py

```diff
@@ -0,0 +1,23 @@
+"""Tests for pip._internal.messages
+"""
+
+import pytest
+
+from pip._internal.utils.messages import oxford_comma_join
+
+
+@pytest.mark.parametrize("test_input,expected",
+                         [('', []),
+                          ('a', ['a']),
+                          ('a and b', ['a', 'b']),
+                          ('a, b, and c', ['a', 'b', 'c'])])
+def test_oxford_comma_join_implicit_conjunction(test_input, expected):
```

Comment:
The parametrized values aren't aligned to the arguments, they are swapped
```suggestion
@pytest.mark.parametrize("test_input, expected",
                         [([], ''),
                          (['a'], 'a'),
                          (['a', 'b'], 'a and b'),
                          (['a', 'b', 'c'], 'a, b, and c')])
def test_oxford_comma_join_implicit_conjunction(test_input, expected):
```